Critical Bug Fix fixed image fusion on mac

Nowhere4Nothing opened the pull request:
Fixed the not loading with mac, used python threading
Fixed auto fusion not loading

## Summary by Sourcery

Add threaded loading and progress queue to Image Fusion UI to fix macOS loading issues and improve error handling

New Features:
- Introduce a progress queue polled by ImageFusionProgressWindow via QTimer
- Enable manual fusion loading in a background Python thread with interrupt flag and progress queue

Bug Fixes:
- Fix manual fusion loading hang on macOS by moving loader into a separate thread with proper progress reporting
- Fix auto fusion branch not loading by handling boolean result types and showing error dialogs
- Replace commonprefix with commonpath in MovingImageLoader to correctly compute shared directory

Enhancements:
- Unify progress_callback usage in ManualFusionLoader and MovingImageLoader as direct callback calls
- Add robust exception handling in MovingImageLoader.load with helper method to emit progress and errors
- Connect fusion mode radio buttons to revalidate and toggle the Confirm button upon mode changes

mattbrgn committed:
Swap image fusion threading to native python

- Qt threading causing bus errors in MacOS environments for Image Fusion loading

Nowhere4Nothing committed:
Update fixed auto fusion

Fixed auto to work with the new manual fusion / python thread or whenever it broke last

Fixed bug: where user could bypass the warning that stopped users doing auto fusion without an rstruct file selected.

Nowhere4Nothing committed:
Update Manual fusion loader

Removed traceback, as its not needed anymore

Kahreiru commented on src/View/ImageFusion/ManualFusionLoader.py:
```diff
@@ -195,14 +204,15 @@ def _load_with_vtk(self, progress_callback):
-            progress_callback.emit(("Preparing overlays...", 90))
+            progress_callback(("Preparing overlays...", 90))
             QtCore.QCoreApplication.processEvents()
+            time.sleep(0.05)
```
while it is only 0.05 seconds. Is this a thread blocking? could a non-thread blocking option be used here

Nowhere4Nothing replied:
it doesn't need to be there at all it just goes too quick and would go 50% to done, was just thinking it would be better for the user to see it go to 90% for a very quick time before loading. can be removed if best

Kahreiru reviewed (approved):
look ok provided things which are mentioned but not a big deal

mattbrgn committed:
Threading safety net

- Added checks to see if progress callback is from a Qt signal and will change the method execution.

mattbrgn committed:
fix autofusion crash after manual

sjswerdloff reviewed (approved):
I created an issue for the renaming of the helper method.
While the bot might consider this to be a nit, this type of thing prevents abstraction.
In other words, I can't read the code calling it and have any idea of what's going to happen, so I have to go *find* the function and read it.  So it doesn't really help me as someone who needs to understand the code, you might as well have just copy/pasted it where-ever it was being used.
